x-pack/plugins/cases/server/client/attachments/bulk_get.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { SavedObject, SavedObjectReference } from '@kbn/core/server';
+
+import Boom from '@hapi/boom';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { fold } from 'fp-ts/lib/Either';
+import { identity } from 'fp-ts/lib/function';
+
+import { partition } from 'lodash';
+import { CASE_SAVED_OBJECT, MAX_BULK_GET_ATTACHMENTS } from '../../../common/constants';
+import type { BulkGetAttachmentsResponse, CommentAttributes } from '../../../common/api';
+import {
+  excess,
+  throwErrors,
+  BulkGetAttachmentsResponseRt,
+  BulkGetAttachmentsRequestRt,
+} from '../../../common/api';
+import { flattenCommentSavedObjects } from '../../common/utils';
+import { createCaseError } from '../../common/error';
+import type { CasesClientArgs, SOWithErrors } from '../types';
+import { Operations } from '../../authorization';
+import type { BulkGetArgs } from './types';
+import type { BulkOptionalAttributes, OptionalAttributes } from '../../services/attachments/types';
+
+type AttachmentSavedObjectWithErrors = SOWithErrors<CommentAttributes>;
+
+type AttachmentSavedObject = SavedObject<CommentAttributes>;
+
+/**
+ * Retrieves multiple attachments by id.
+ */
+export async function bulkGet(
+  { attachmentIDs, caseID }: BulkGetArgs,
+  clientArgs: CasesClientArgs
+): Promise<BulkGetAttachmentsResponse> {
+  const {
+    services: { attachmentService },
+    logger,
+    authorization,
+  } = clientArgs;
+
+  try {
+    const request = pipe(
+      excess(BulkGetAttachmentsRequestRt).decode({ ids: attachmentIDs }),
+      fold(throwErrors(Boom.badRequest), identity)
+    );
+
+    throwErrorIfIdsExceedTheLimit(request.ids);
+
+    const attachments = await attachmentService.getter.bulkGet(request.ids);
```

Comment:
Do you think we should `await casesClient.cases.resolve` before the call to `bulkGet` to ensure that we have authorization for reading the case?